src/client/java/net/neoforged/neoforge/client/resources/NeoSplashManager.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.resources;
+
+import static java.time.temporal.ChronoField.DAY_OF_MONTH;
+import static java.time.temporal.ChronoField.MONTH_OF_YEAR;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.mojang.logging.LogUtils;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.JsonOps;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import java.io.IOException;
+import java.io.Reader;
+import java.time.Month;
+import java.time.MonthDay;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import net.minecraft.client.User;
+import net.minecraft.client.gui.components.SplashRenderer;
+import net.minecraft.client.resources.SplashManager;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.Resource;
+import net.minecraft.server.packs.resources.ResourceManager;
+import net.minecraft.util.GsonHelper;
+import net.minecraft.util.profiling.ProfilerFiller;
+import org.jetbrains.annotations.Nullable;
+
+public class NeoSplashManager extends SplashManager {
```

Comment:
I'm a bit confused. Why is this its own class? It seems like this could be done by just, like, a `NeoSplashHooks` class or something and then a patch in the vanilla `SplashManager` which would avoid both the needless ATs _and_ the need to remove a vanilla listener and have a neo listener instead.

Patching would _also_ make it clear to people that the original logic in `SplashManager` -- an if-else chain for the special dates and all -- is actually completely gone, instead of basically "silently" not calling it as here.